### PR TITLE
Add option `collapsedHeaderBackground` for custom `CollapsedHeaderBackground`

### PR DESCRIPTION
**Pull request by beata**

### CODE REVIEW

**Comment by benevbright** [comment]
Hi, @beata Thanks.
Could you provide an example when we need this or what's the current problem?

**Comment by beata** [comment]
Hi, @benevbright,

I want to make the header background styled with box-shadow and border and use a common background color.

In react-navigation-collapsible v5, the only configurable option in `CollapsedHeaderBackground` is the header background, and the background color must be specified in each screen.

In my case, all screens are sharing the same screen options from the navigator, so the `backgroundColor` in `CollapsedHeaderBackground` would always be null.


Here' an example of how I use the new option to apply the custom header background

Navigation Stack:
```typescript
export const collapsibleOptions: CollapsibleStackConfig = {
  collapsedColor: paper.default.colors.transparent,
  collapsedHeaderBackground: AppHeaderBackground,
}

export function HomeStack() {
  return (
    <Home.Navigator headerMode={headerMode} screenOptions={screenOptions}>
      {createCollapsibleStack(
        <Home.Screen name="Home" component={HomeScreen} />,
        collapsibleOptions,
      )}
    </Home.Navigator>
  )
}
```
AppHeaderBackground:
```typescript
import React from "react"
import { Animated, View, StyleSheet, Platform } from "react-native"
import { screenOptions } from "./navigation-options"

const CONTAINER = { flex: 1 }
const STATIC_BG = { position: "absolute", width: "100%", height: "100%" }
const ANIAMTED_BG = {
  flex: 1,
  ...Platform.select({
    android: {
      elevation: 4,
    },
    ios: {
      shadowOpacity: 0.85,
      shadowRadius: 0,
      shadowOffset: {
        width: 0,
        height: StyleSheet.hairlineWidth,
      },
    },
    default: {
      borderBottomWidth: StyleSheet.hairlineWidth,
    },
  }),
}

export const AppHeaderBackground = ({
  translateY,
  opacity,
  backgroundColor = screenOptions.headerStyle?.backgroundColor,
  collapsedColor,
}) => () => (
  <Animated.View style={[CONTAINER, { transform: [{ translateY }] }]}>
    <View style={[STATIC_BG, { backgroundColor: collapsedColor || backgroundColor }]} />
    <Animated.View
      style={[
        ANIAMTED_BG,
        {
          backgroundColor,
          opacity,
        },
      ]}
    />
  </Animated.View>
)
```

File: src/types.ts [hunk 1]
```diff
@@ -11,4 +12,5 @@ export type Collapsible = {
 
 export type CollapsibleStackConfig = {
   collapsedColor?: string;
+  collapsedHeaderBackground?: ReactNode;
```
Comment (benevbright) [inline src/types.ts:15]: How about naming `CollapsedHeaderBackground` so we know it's a component?

Comment (beata) [reply src/types.ts]: It would be great.
I've pushed another commit for this change.

**Comment by benevbright** [comment]
There was a type error and I fixed it https://github.com/benevbright/react-navigation-collapsible/commit/077830ab024a31c7aac10004829e01529cbb65dc
This PR is included in `v5.3.0`. Thanks again!

**Comment by benevbright** [comment]
Actually, it was not a component. I was confused for a moment. 
Maybe capitalizing was not a good idea. 😁

**Pull request merged**